vllm/model_executor/models/aria.py

```diff
@@ -390,14 +388,18 @@ def load_weights(self, weights: Iterable[Tuple[str,
                 # Models trained using ColossalAI may include these tensors in
                 # the checkpoint. Skip them.
                 continue
-            if scale_name := get_compressed_tensors_cache_scale(name):
-                # Loading kv cache scales for compressed-tensors quantization
-                param = params_dict[scale_name]
-                weight_loader = getattr(param, "weight_loader",
-                                        default_weight_loader)
-                loaded_weight = loaded_weight[0]
-                weight_loader(param, loaded_weight)
-                loaded_params.add(scale_name)
+            if (self.quant_config is not None and
+                (scale_names := self.quant_config.get_cache_scale(name))):
+                # Loading kv cache scales for quark and
+                # compressed-tensors quantization
+                for scale_name in scale_names:
+                    param = params_dict[scale_name]
+                    weight_loader = getattr(param, "weight_loader",
+                                            default_weight_loader)
+                    loaded_weight = (loaded_weight if loaded_weight.dim() == 0
+                                     else loaded_weight[0])
+                    weight_loader(param, loaded_weight)
+                    loaded_params.add(scale_name)
```

Comment:
Like I said before, why does `get_cache_scale` need to return a list? This would be simpler if it was a 1-to-1 function.

Also what is the purpose of the loaded_weight condition? I actually think it would be better to treat it as a tensor in the long run rather than a scalar, so we can move beyond per-tensor in the future. This is out of scope for this PR though

Reply:
In quark's model, if it is per_tensor quantization, the corresponding scale is a scalar expressed as torch.tensor(scale). Using tensor[0] to obtain it will report an error, so we added a judgment. This will also increase the robustness of the system to prevent scale from being a scale.

Reply:
Okay, we will consider better ways to solve this problem in the future.